### ISSUE TITLE
Scheduler seems to be vectorizing size-0 tensor, resulting in validation failure

### ISSUE THREAD

**Issue by jjsjann123** [csarofeen/pytorch#1446]
### 🐛 Describe the bug

Error extracted from failures encountered in https://github.com/pytorch/pytorch/pull/71299

Looks like the scheduler are vectorizing a size-0 tensor. It is strange that when I update input size from `(1, 0, 3)` to `(3, 0, 3)` and the vectorization went away :confused: . In my naive understanding, neither sizes are vectorization-friendly...

```
root@992e27066e89:/raid/playground# PYTORCH_JIT_LOG_LEVEL=graph_fuser PYTORCH_NVFUSER_DISABLE_FALLBACK=1 PYTORCH_NVFUSER_DUMP=cuda_kernel python opinfo_failure_3.py
[DUMP graph_fuser.cpp:2236] Before Fusion:
[DUMP graph_fuser.cpp:2236] graph(%t0.1 : Tensor):
[DUMP graph_fuser.cpp:2236]   %3 : float = prim::Constant[value=0.5]() # opinfo_failure_3.py:13:12
[DUMP graph_fuser.cpp:2236]   %2 : float = prim::Constant[value=1.]() # opinfo_failure_3.py:12:12
[DUMP graph_fuser.cpp:2236]   %1 : int = prim::Constant[value=1]()
[DUMP graph_fuser.cpp:2236]   %4 : Tensor = prim::profile[profiled_type=Float(1, 0, 3, strides=[3, 3, 1], requires_grad=0, device=cuda:0)](%t0.1)
[DUMP graph_fuser.cpp:2236]   %t1.1 : Tensor = aten::add(%4, %2, %1) # opinfo_failure_3.py:12:7
[DUMP graph_fuser.cpp:2236]   %6 : Tensor = prim::profile[profiled_type=Float(1, 0, 3, strides=[3, 3, 1], requires_grad=0, device=cuda:0)](%t1.1)
[DUMP graph_fuser.cpp:2236]   %t2.1 : Tensor = aten::mul(%6, %3) # opinfo_failure_3.py:13:7
[DUMP graph_fuser.cpp:2236]   %8 : Tensor = prim::profile[profiled_type=Float(1, 0, 3, strides=[3, 3, 1], requires_grad=0, device=cuda:0)](%t2.1)
[DUMP graph_fuser.cpp:2236]   %9 : Tensor = aten::relu(%8) # opinfo_failure_3.py:14:9
[DUMP graph_fuser.cpp:2236]   %10 : Tensor = prim::profile[profiled_type=Float(1, 0, 3, strides=[3, 3, 1], requires_grad=0, device=cuda:0)](%9)
[DUMP graph_fuser.cpp:2236]   return (%10)
[DUMP graph_fuser.cpp:2310] Before Compilation:
[DUMP graph_fuser.cpp:2310] graph(%t0.1 : Tensor):
[DUMP graph_fuser.cpp:2310]   %15 : bool = prim::CudaFusionGuard[types=[Float(1, 0, 3, strides=[3, 3, 1], requires_grad=0, device=cuda:0)]](%t0.1)
[DUMP graph_fuser.cpp:2310]   %14 : Tensor = prim::If(%15)
[DUMP graph_fuser.cpp:2310]     block0():
[DUMP graph_fuser.cpp:2310]       %11 : Tensor = prim::CudaFusionGroup_0(%t0.1)
[DUMP graph_fuser.cpp:2310]       -> (%11)
[DUMP graph_fuser.cpp:2310]     block1():
[DUMP graph_fuser.cpp:2310]       %22 : Tensor = prim::FallbackGraph_1(%t0.1)
[DUMP graph_fuser.cpp:2310]       -> (%22)
[DUMP graph_fuser.cpp:2310]   return (%14)
[DUMP graph_fuser.cpp:2310] with prim::CudaFusionGroup_0 = graph(%5 : Float(1, 0, 3, strides=[3, 3, 1], requires_grad=0, device=cuda:0)):
[DUMP graph_fuser.cpp:2310]   %3 : float = prim::Constant[value=0.5]() # opinfo_failure_3.py:13:12
[DUMP graph_fuser.cpp:2310]   %7 : int = prim::Constant[value=1]()
[DUMP graph_fuser.cpp:2310]   %6 : float = prim::Constant[value=1.]() # opinfo_failure_3.py:12:12
[DUMP graph_fuser.cpp:2310]   %t1.1 : Float(1, 0, 3, strides=[3, 3, 1], requires_grad=0, device=cuda:0) = aten::add(%5, %6, %7) # opinfo_failure_3.py:12:7
[DUMP graph_fuser.cpp:2310]   %t2.1 : Float(1, 0, 3, strides=[3, 3, 1], requires_grad=0, device=cuda:0) = aten::mul(%t1.1, %3) # opinfo_failure_3.py:13:7
[DUMP graph_fuser.cpp:2310]   %1 : Float(1, 0, 3, strides=[3, 3, 1], requires_grad=0, device=cuda:0) = aten::relu(%t2.1) # opinfo_failure_3.py:14:9
[DUMP graph_fuser.cpp:2310]   return (%1)
[DUMP graph_fuser.cpp:2310] with prim::FallbackGraph_1 = graph(%t0.1 : Float(1, 0, 3, strides=[3, 3, 1], requires_grad=0, device=cuda:0)):
[DUMP graph_fuser.cpp:2310]   %1 : float = prim::Constant[value=0.5]() # opinfo_failure_3.py:13:12
[DUMP graph_fuser.cpp:2310]   %2 : int = prim::Constant[value=1]()
[DUMP graph_fuser.cpp:2310]   %3 : float = prim::Constant[value=1.]() # opinfo_failure_3.py:12:12
[DUMP graph_fuser.cpp:2310]   %t1.1 : Float(1, 0, 3, strides=[3, 3, 1], requires_grad=0, device=cuda:0) = aten::add(%t0.1, %3, %2) # opinfo_failure_3.py:12:7
[DUMP graph_fuser.cpp:2310]   %t2.1 : Float(1, 0, 3, strides=[3, 3, 1], requires_grad=0, device=cuda:0) = aten::mul(%t1.1, %1) # opinfo_failure_3.py:13:7
[DUMP graph_fuser.cpp:2310]   %6 : Float(1, 0, 3, strides=[3, 3, 1], requires_grad=0, device=cuda:0) = aten::relu(%t2.1) # opinfo_failure_3.py:14:9
[DUMP graph_fuser.cpp:2310]   return (%6)
Traceback (most recent call last):
  File "opinfo_failure_3.py", line 20, in <module>
    outs = f(x)
RuntimeError: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript (most recent call last):
RuntimeError: Tried to vectorize a dim resulting in a word size of 0 however, vector sizes only upto and including 16 bytes are supported.
```

Python scripts below:

```
import torch                                                                                                               
import torch.nn as nn                                                                                                      
from torch import Tensor                                                                                                   
import time                                                                                                                
                                                                                                                           
from typing import List                                                                                                    
                                                                                                                           
torch._C._jit_set_nvfuser_single_node_mode(True)                                                                           
                                                                                                                           
@torch.jit.script                                                                                                          
def f(t0):                                                                                                                 
  t1 = t0 + 1.0                                                                                                            
  t2 = t1 * 0.5                                                                                                            
  return t2.relu()                                                                                                         
                                                                                                                           
#x = torch.randn(3, 0, 3, device='cuda')  # changing the size apparently gets the error away                                                                                   
x = torch.randn(1, 0, 3, device='cuda')                                                                                    
with torch.jit.fuser("fuser2"):                                                                                                                                                                                    
  for i in range(5):                                                                                                       
    outs = f(x)                                                                                                            
  print(f.graph_for(x))                                                                                                    
  print(outs)
```

### Versions

commit 16df2b8fa20c446766e27417e0a9d565b1784853 (HEAD, csarofeen/devel)
Author: Naoya Maruyama <naoyam@users.noreply.github.com>
Date:   Tue Feb 8 15:21:47 2022 -0800

    print 0-dim tensors as tensors (#1442)

**Comment by jjsjann123** [comment]
Not sure if I'm tagging the right person, but I remember @rdspring1 working on vectorization before so there you go :pray:

**Comment by rdspring1** [comment]
_Maybe_ fixed by #1459
